stablehlo/dialect/TypeInference.cpp

```diff
@@ -3625,15 +3619,17 @@ LogicalResult verifyReduceOp(std::optional<Location> location,
   SmallVector<ShapedType> initValueTypes{llvm::map_range(
       initValues.getTypes(), [](Type t) { return t.cast<ShapedType>(); })};
 
-  // P1. & P2.
+  if (dimensions.getType().getRank() != 1)
+    return emitOptionalError(location, "dimensions must be rank 1");
+
   SmallVector<int64_t> newDimensions;
   Attribute encoding;
+  // reduce_c1, reduce_c4, reduce_c5
```

Comment:
Let's leave the same comment on the second usage of verifyReduceOpInputsAndInferShape?